src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -9,11 +9,144 @@
 
 package common
 
-import "fmt"
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/cmd/container/common"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"golang.org/x/sync/errgroup"
+)
 
 const FeedbackMsg = `We'd love to hear about your experience with redpanda:
 https://vectorized.io/feedback`
 
 func DeprecationMessage(altCmd string) string {
 	return fmt.Sprintf("use '%s' instead.", altCmd)
 }
+
+// Try to read the config from the default expected locations, or from the
+// specific path passed with --config. If --config wasn't passed, and the config
+// wasn't found, return the default configuration.
+func FindConfigFile(
+	mgr config.Manager, configFile *string,
+) func() (*config.Config, error) {
+	var conf *config.Config
+	var err error
+	return func() (*config.Config, error) {
+		if conf != nil {
+			return conf, nil
+		}
+		conf, err = mgr.ReadOrFind(*configFile)
+		if err != nil {
+			log.Debug(err)
+			if os.IsNotExist(err) && *configFile == "" {
+				log.Debug(
+					"Config file not found and --config" +
+						" wasn't passed, using default" +
+						" config",
+				)
+				return config.Default(), nil
+			}
+		}
+		return conf, err
+	}
+}
+
+func ContainerBrokers() []string {
+	c, err := common.NewDockerClient()
+	if err != nil {
+		log.Debug(err)
+		return []string{}
```

Comment:
I'm not a golang dude (again), but if these functions are returning value, error tuples, can you return the same thing? Instead of empty string list, return some error?

Reply:
This is just a part of a chain of possible broker addresses sources. It goes
1. Look for a local container cluster (i.e. one started with `rpk container start`)
2. Look for a local config file
3. Use the default config (addr = 127.0.0.1:9092)

So if this fails, or if the resulting addrs list is empty, the next step is tried and so on. The errors aren't fatal.

Reply:
ah i see, sweet!